### PR TITLE
Feature/unit test for find

### PR DESCRIPTION
**Pull request by phanletrunghieu**
- [x] unit test FindAll
- [x] unit test FindByColumnName
- [x] config data test

### CODE REVIEW

File: backend/sqlmapper/drivers/pg_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	"strings"
+	"testing"
+
+	backendConfig "github.com/dwarvesf/smithy/backend/config"
+	"github.com/dwarvesf/smithy/backend/sqlmapper"
+	"github.com/dwarvesf/smithy/common/database"
+	"github.com/dwarvesf/smithy/common/utils"
+)
+
+func beforeTest(t *testing.T) *backendConfig.Config {
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: Change this to agent integration test

File: backend/sqlmapper/drivers/pg_test.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	if err != nil {
+		t.Fatalf("Can't read config file %v", err)
+	}
+
+	err = cfg.UpdateConfigFromAgent()
+	if err != nil {
+		t.Fatalf("Can't sync from agent %v", err)
+	}
+
+	return cfg
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: We should use config by change value instead of make a connection to agent

File: backend/sqlmapper/drivers/pg_test.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	sampleData["id"] = sqlmapper.ColData{
+		Data:     999,
+		DataType: "int",
+	}
+	sampleData["name"] = sqlmapper.ColData{
+		Data:     "hieudeptrai",
+		DataType: "string",
+	}
+
+	tableName := "users"
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: should not use auto migrate from agent

File: backend/sqlmapper/drivers/pg_test.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	if err != nil {
+		t.Fatalf("Can't sync from agent %v", err)
+	}
+
+	return cfg
+}
+
+func Test_pgStore_FindAll(t *testing.T) {
+	// read yaml & sync config from agent
+	cfg := beforeTest(t)
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: check example-go to make test_util to make clean database after test

File: common/utils/utils.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package utils
```
Comment (hieunmce) [inline common/utils/utils.go]: should specify name for utils and move to another package (ex: relfect)

File: makefile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-.PHONY: build up-agent up-dashboard local-db
+.PHONY: build up-agent up-agent-test up-dashboard local-db
```
Comment (hieunmce) [inline makefile]: should make test cmd in here

File: agent/config/yaml.go [hunk 7]
```diff
@@ -26,5 +27,14 @@ func (c yamlReaderImpl) Read() (*Config, error) {
 		return nil, err
 	}
 
+	if os.Getenv("ENV") == "test" {
```
Comment (hieunmce) [inline agent/config/yaml.go]: Should not use this

File: backend/sqlmapper/drivers/pg_test.go [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+	"strings"
+	"testing"
+
+	"github.com/dwarvesf/smithy/backend/sqlmapper"
+	"github.com/dwarvesf/smithy/common/database"
+	utilReflect "github.com/dwarvesf/smithy/common/utils/reflect"
+	utilTest "github.com/dwarvesf/smithy/common/utils/test"
+)
+
+// func Test_SyncAgent(t *testing.T) {
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: We should add comment in here

File: common/utils/test/test.go [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+const (
+	dbHost     = "localhost"
+	dbPort     = "5439"
+	dbUser     = "postgres"
+	dbPassword = "example"
+	dbName     = "test"
+)
+
+// MigrateTables migrate db with tables base by domain model
+func MigrateTables(db *gorm.DB) error {
```
Comment (hieunmce) [inline common/utils/test/test.go]: We should remove this function or have a better name

File: common/utils/test/test.go [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+	type User struct {
+		Id   int `sql:"primary_key"`
+		Name string
+	}
+	return db.AutoMigrate(
+		User{},
+	).Error
+}
+
+func CreateModelList() []database.Model {
```
Comment (hieunmce) [inline common/utils/test/test.go]: We should have better name for this function

File: common/utils/test/test.go [hunk 11]
```diff
@@ -0,0 +1,111 @@
+package test
```
Comment (hieunmce) [inline common/utils/test/test.go]: should have better name

File: backend/sqlmapper/drivers/pg_test.go [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+		Data:     999,
+		DataType: "int",
+	}
+	sampleData["name"] = sqlmapper.ColData{
+		Data:     "hieudeptrai",
+		DataType: "string",
+	}
+
+	tableName := "users"
+	cfg.DB().Exec(fmt.Sprintf("DELETE FROM %s;", tableName))
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: remove deleted table

File: backend/sqlmapper/drivers/pg_test.go [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: remove deleted table

File: backend/sqlmapper/drivers/pg_test.go [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+		want    []sqlmapper.RowData
+		wantErr bool
+	}{
+		{
+			name: "Valid test case",
+			args: &args{
+				Offset: 0,
+				Limit:  10,
+			},
+			want: []sqlmapper.RowData{sampleData},
```
Comment (hieunmce) [inline backend/sqlmapper/drivers/pg_test.go]: need a test case for empty,
test case using limit and offset
